### PR TITLE
Refactor ModelQuery

### PR DESCRIPTION
**Pull request by digimaun**
- Refactor ModelQuery to simplify, improve readability and maintainability

### CODE REVIEW

File: sdk/modelsrepository/Azure.Iot.ModelsRepository/src/ModelPropertyConstants.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Iot.ModelsRepository
+{
+    internal static class ModelPropertyConstants
```
Comment (drwill-ms) [inline sdk/modelsrepository/Azure.Iot.ModelsRepository/src/ModelPropertyConstants.cs:6]: class comments for future code owners. What belongs in this class and why?

File: sdk/modelsrepository/Azure.Iot.ModelsRepository/src/ModelQuery.cs [hunk 2]
```diff
@@ -42,93 +49,132 @@ public string GetId()
             return string.Empty;
         }
 
-        public IList<string> GetExtends()
+        private static IList<string> ParseExtends(JsonElement root)
         {
             List<string> dependencies = new List<string>();
```
Comment (drwill-ms) [inline sdk/modelsrepository/Azure.Iot.ModelsRepository/src/ModelQuery.cs:54]: var as variable type, as type is obvious from initialization

File: sdk/modelsrepository/Azure.Iot.ModelsRepository/src/ModelQuery.cs [hunk 3]
```diff
@@ -42,93 +49,132 @@ public string GetId()
...
+
+        private static bool isInterfaceObject(JsonElement root)
+        {
+            return root.ValueKind == JsonValueKind.Object &&
+                root.TryGetProperty(ModelPropertyConstants.Type, out JsonElement objectType) &&
+                objectType.ValueKind == JsonValueKind.String &&
+                objectType.GetString() == ModelPropertyConstants.TypeValueInterface;
+        }
+
+        private static bool isComponentObject(JsonElement root)
```
Comment (drwill-ms) [inline sdk/modelsrepository/Azure.Iot.ModelsRepository/src/ModelQuery.cs:172]: Even private methods in .NET follow TitleCasing

**Review by drwill-ms — APPROVED** [review]

File: sdk/modelsrepository/Azure.Iot.ModelsRepository/src/ModelPropertyConstants.cs (hunk 1 above)
Comment (azabbasi) [reply sdk/modelsrepository/Azure.Iot.ModelsRepository/src/ModelPropertyConstants.cs:10]: There is a class called ModelRepositoryConstants, we can have these constants in that file as well. even if they are in their own nested subclass.

**Review by azabbasi — APPROVED** [review]

File: sdk/modelsrepository/Azure.Iot.ModelsRepository/src/ModelPropertyConstants.cs (hunk 1 above)
Comment (digimaun) [reply sdk/modelsrepository/Azure.Iot.ModelsRepository/src/ModelPropertyConstants.cs]: Made it so.

**Pull request merged**